src/main/java/io/kestra/plugin/ldapManager/LdapConnection.java

```diff
@@ -0,0 +1,60 @@
+package io.kestra.plugin.ldapManager;
+
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPException;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.Task;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+import jakarta.validation.constraints.NotNull;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+@SuperBuilder
+@Getter
+@NoArgsConstructor
+public abstract class LdapConnection extends Task {
+    @Schema(
+        title = "Hostname",
+        description = "Hostname for connection."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    protected String hostname;
+
+    @Schema(
+        title = "Port",
+        description = "Port for connection."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    protected Integer port;
+
+    @Schema(
+        title = "User DN",
+        description = "User DN for connection."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    protected String userDn;
+
+    @Schema(
+        title = "Password",
+        description = "User password for connection."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    protected String password;
```

Comment:
A dynamic plugin's property can accept a pebble expression and must be rendered via one of the `runContext#render` methods. This is particularly important for sensitive properties (such as passwords), as it allows such properties to not be defined in plain text in the YAML definition of the Flow.

In your context, before creating the `LDAPConnection` you need to render each plugin properties manually.

e.g.

```java
return new LDAPConnection(
   runContext().render(hostname), 
   runContext().render(port), 
   runContext().render(userDn), 
   runContext().render(password)
);
```